src/deploy/functions/runtimes/node/parseTriggers.ts

```diff
@@ -302,7 +302,7 @@ export function addResourcesToBuild(
     project: projectId,
     entryPoint: annotation.entryPoint,
     runtime: runtime,
-    serviceAccount: annotation.serviceAccountEmail || "default",
+    serviceAccount: annotation.serviceAccountEmail || null,
```

Comment:
It's quite possible that `undefined` will trigger different behavior than `null`. They behave differently in `JSON.stringify` and `proto.fieldMasks` includes nulls but not undefined too. This likely has the side effect of resetting service accounts on every deploy to their default.